File: Javascript/BinarySearch.js
let min = 1;
let max = 1000000;
let lowerBound = min;
let higherBound = max;

let number = Math.floor(Math.random() * max) + min;

let count = 1;
function binarySearch(guess) {
  if (guess > number) {
    higherBound = guess;
    count++;
    binarySearch(Math.round((lowerBound + higherBound) / 2));
  } else if (guess < number) {
    lowerBound = guess;
    count++;
    binarySearch(Math.round((lowerBound + higherBound) / 2));
  } else if (guess == number) {
    console.log("The answer is " + guess + ".");
    console.log("This took " + count + " tries.");
  }
}

binarySearch(Math.round(max / 2));

File: BinarySearch.py
from random import randint

min = 1
max = 100000000
lowerBound = min
higherBound = max

number = randint(min, max)

count = 1


def binarySearch(guess):
    global count
    global lowerBound
    global higherBound
    if (guess > number):
        higherBound = guess
        count += 1
        binarySearch(round((lowerBound + higherBound) / 2))
    elif (guess < number):
        lowerBound = guess
        count += 1
        binarySearch(round((lowerBound + higherBound) / 2))
    elif (guess == number):
        print("The answer is " + str(int(guess)) + ".")
        print("This took " + str(count) + " tries.")


binarySearch(round(max / 2))
